### ISSUE TITLE
Linker error due to relocation issue in assembly files

### ISSUE THREAD

**Issue by Tikrass** [pascholl/SimpleOT#2]
Hello, 
I would like to use [SPDZ2](https://github.com/bristolcrypto/SPDZ-2) in my research which will be used in my master thesis at RWTH Aachen. The MASCOT extension relies on this SimpleOT protocol. Unfortunately I am not able to compile and link the code to a shared object. 

I tried to compile it with `make clean all`. But i get many like the following errors from the linker:
>/usr/bin/ld: fe25519_freeze.o: relocation R_X86_64_32S against symbol `CONST_REDMASK51' can not be used when making a shared object; recompile with -fPIC

[Here](https://github.com/pascholl/SimpleOT/files/1207948/make.txt) is the console output.

It seems that there is some problem with the relocation, which comes from loading constant values from the data section. Maybe it has to do something with using `movabs` instead of `mov` but I am not experienced enough with assembly language to solve the problem by myself. 

Can you help me, how to build this library? Thanks in advance!

Here are the versions of my used tools:
> $ ld --version
> GNU ld (GNU Binutils) 2.28.0.20170506

> $ gcc --version
> gcc (GCC) 7.1.1 20170630

> $ as --version
> GNU assembler (GNU Binutils) 2.28.0.20170506

I try to build for 64bit architecture on Arch Linux.

**Comment by pascholl** [comment]
Hi Tikras,

I did not write this software so I'm not sure if I'll be able to help (you
may want to also contact Tung Chou, the original SimpleOT author).

I'm not sure why that error would occur, I've tested on GCC versions 4.48
and 6.2 with no problems. Did you try adding -fPIC to CFLAGS in the
makefile as the message suggests?

Also, I think the code relies on SSE instructions of Intel CPUs, so you may
want to check your machine supports those (cat /proc/cpuinfo).

Peter

**Comment by Tikrass** [comment]
It looks like it is a problem with the new versions of gcc and binutils, who 
I was able to compile it, using an old version of gcc (6.3). 

Therefore I changed in the Makefile the first line to 
> CC = /usr/bin/gcc-6.3

The suggestion using the -fPIC option would not help, because it does only affect C/C++ compilation but not the linker and the assembler files, where the error occurs. However, big thanks for your help and time! 

Regards, 
Thibaud
